sensor/common/scannerdefinitions/http_handler.go

```diff
@@ -0,0 +1,79 @@
+package scannerdefinitions
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/utils"
+	"google.golang.org/grpc/codes"
+)
+
+// scannerDefinitionsHandler handle requests to retrieve scanner vulnerability
+// definitions from Central.
+type scannerDefinitionsHandler struct {
+	http.Handler
+	centralClient *http.Client
+	centralHost   string
+}
+
+// NewDefinitionsHandler creates a new scanner definitions handler.
+func NewDefinitionsHandler(centralHost string) (http.Handler, error) {
+	client, err := clientconn.NewHTTPClient(mtls.CentralSubject, centralHost, 0)
```

Comment:
I think we should have a timeout here. I do wonder why we don't have one for the kocache...

Reply:
Do we need to? Sensor waits until Central gives up, and not sooner than the client requested. I don't think the Context has anything to do with it, let m know.

Reply:
From what I can tell, the [server-side's request context is canceled when the client closes the connection](https://pkg.go.dev/net/http#Request.Context), which I'm guessing happens upon timeout. This makes me believe the client-side timeout is directly related to the server-side's request, so it is good that you do

```
centralRequest, err := http.NewRequestWithContext(request.Context(), http.MethodGet, centralURL.String(), nil)
```

This ensures that context is passed along and we should respect the original client's timeout